crates/ruff_cli/src/commands/rule.rs

```diff
@@ -11,73 +13,102 @@ use crate::args::HelpFormat;
 #[derive(Serialize)]
 struct Explanation<'a> {
     name: &'a str,
-    code: &'a str,
+    code: String,
     linter: &'a str,
     summary: &'a str,
     message_formats: &'a [&'a str],
-    autofix: &'a str,
+    autofix: String,
     explanation: Option<&'a str>,
 }
 
-/// Explain a `Rule` to the user.
-pub(crate) fn rule(rule: Rule, format: HelpFormat) -> Result<()> {
-    let (linter, _) = Linter::parse_code(&rule.noqa_code().to_string()).unwrap();
-    let mut stdout = BufWriter::new(io::stdout().lock());
-    let mut output = String::new();
+fn rule_to_explanation(rule: &Rule) -> Explanation<'_> {
+    let code = rule.noqa_code().to_string();
+    let (linter, _) = Linter::parse_code(&code).unwrap();
+    let autofix = rule.autofixable().to_string();
+    Explanation {
+        name: rule.as_ref(),
+        code,
+        linter: linter.name(),
+        summary: rule.message_formats()[0],
+        message_formats: rule.message_formats(),
+        autofix,
+        explanation: rule.explanation(),
+    }
+}
 
-    match format {
-        HelpFormat::Text => {
-            output.push_str(&format!("# {} ({})", rule.as_ref(), rule.noqa_code()));
-            output.push('\n');
-            output.push('\n');
+fn format_rule_text(rule: Rule) -> String {
```

Comment:
Nit: You could instead also implement `Display` for e.g. `Explanation` (this would also ensure that the JSON output and the human-readable output contain the same information).

Reply:
I think I'll leave that for a future refactoring (I think the docs-generation code could be unified with this too?).